src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -440,6 +440,18 @@ void RayletBasedActorScheduler::HandleWorkerLeaseReply(
   // gcs_actor_manager will reconstruct it again.
   auto node_id = NodeID::FromBinary(node->node_id());
   auto iter = node_to_actors_when_leasing_.find(node_id);
+  auto kill_worker = [&reply, this]() {
+    auto &worker_address = reply.worker_address();
+    if (!worker_address.raylet_id().empty()) {
+      auto cli = core_worker_clients_.GetOrConnect(worker_address);
+      rpc::KillActorRequest request;
+      // Set it to be Nil() since it hasn't been setup yet
+      request.set_intended_actor_id(ActorID::Nil().Binary());
+      request.set_force_kill(true);
+      request.set_no_restart(true);
+      RAY_UNUSED(cli->KillActor(request, nullptr));
```

Comment:
This RPC will fail most of time right? Wonder if we can have DEBUG log here for the future debugging (in case this introduces any bug)

Reply:
What do you mean here by "fail"? Is it because it's killing the actor? I think it depends on the implementation of client-side code. Ideally, it shouldn't.

Reply:
when I say fail, I meant the RPC returns the error status because the actor is already killed. So, my hypothesis here is that most of time actors are already killed, and in some situations, they are not killed, and this code will ensure the correct "final state". Did I understand this part correctly? I'd like to add logs in this case. 

```cpp
 RAY_UNUSED(cli->KillActor(request, [](Status &status) { RAY_LOG(DEBUG) << "Killing an actor because the lease request is returned after the actor is killed. Killing status: " << status}));
```